packages/plans-grid-next/src/lib/filter-unused-features-object.ts

```diff
@@ -16,9 +16,11 @@ const filterUnusedFeaturesObject = (
 	const uniqueFeaturesAvailable = new Set(
 		visibleGridPlans
 			.map( ( gridPlan ) =>
-				[ ...gridPlan.features.wpcomFeatures, ...gridPlan.features.jetpackFeatures ].map(
-					( feature ) => feature.getSlug()
-				)
+				[
+					...gridPlan.features.wpcomFeatures,
+					...gridPlan.features.jetpackFeatures,
+					...( gridPlan.features.conditionalFeatures ?? [] ),
```

Comment:
The fix here was to include `conditionalFeatures` alongside `wpcomFeatures` and `jetpackFeatures` so that any 'conditiona filters' aren't excluded from the filtering (eg. in this case `FEATURE_COMMISSION_FEE_WOO_FEATURES` which is [defined](https://github.com/Automattic/wp-calypso/blob/a9d53c622808bb7604a84efb8eadd1330df6a0fc/packages/calypso-products/src/plans-list.tsx#L1115) as a conditional feature).

I'm still quite new to the codebase -- the tests pass however I'd appreciate additional 👀 from any @Automattic/martech-decepticons or perhaps @ilyasfoo who previously worked on this filtering) to verify that this change makes sense.

🙏

Reply:
I think what you are doing here is good enough.

> (eg. in this case FEATURE_COMMISSION_FEE_WOO_FEATURES which is [defined](https://github.com/Automattic/wp-calypso/blob/a9d53c622808bb7604a84efb8eadd1330df6a0fc/packages/calypso-products/src/plans-list.tsx#L1115) as a conditional feature)

I'm curious what "conditional" specifically refers to in `get2023PlanComparisonConditionalFeatures`. Is this akin to "additional" (meaning, "additionally" to something else)?

That whole data features structure is a bit of soup.

Reply:
> That whole plan features structure is a bit of soup.

Yes I've found `packages/calypso-products/src/plans-list.tsx` quite difficult to wrap my head around. I'll keep this in mind as I gather more context, in terms of ideas to help simplify/clarify. Hat tip to @jdc91 for including comments here that helped clarify the 'override' and 'conditional' functions:

https://github.com/Automattic/wp-calypso/blob/fa6317d876bb38c67e76da8e4caf9b92a657440d/packages/calypso-products/src/types.ts#L256-L266

> I'm curious what "conditional" specifically refers to in get2023PlanComparisonConditionalFeatures. Is this akin to "additional" (meaning, "additionally" to something else)?

My understanding is that a 'conditional' feature has attributes that can change based on the plan. The conditions can be seen here in this (optional) `getConditionalTitle` function (I'm not yet aware of other spots where these conditions might be used):

https://github.com/Automattic/wp-calypso/blob/14dad839b94ae077af5a4ca8627fea3522efc759/client/lib/plans/features-list.tsx#L1775-L1791